web/src/edit/NeighbourhoodMode.svelte

```diff
@@ -372,12 +483,48 @@
       <AnimatePaths paths={allShortcuts} />
     {/if}
 
-    <ModalFilterLayer on:click={deleteFilter}>
-      <Popup openOn="hover">Click to delete</Popup>
+    <ModalFilterLayer
+      onClickModalFilter={deleteModalFilter}
+      onClickTurnRestriction={deleteTurnRestriction}
+    >
+      <div slot="modal-filter">
+        <Popup openOn="hover">Click to delete</Popup>
+      </div>
+      <div slot="turn-restriction">
+        <Popup openOn="hover">Click to delete</Popup>
+      </div>
     </ModalFilterLayer>
 
-    {#if action == "freehand-filters"}
+    {#if action.kind == "freehand-filters"}
       <FreehandLine map={notNull($map)} on:done={gotFreehandLine} />
     {/if}
+
+    {#if action.kind == "turn_restriction"}
```

Comment:
![image](https://github.com/user-attachments/assets/d81c21a7-1875-42b9-9f09-2cc850f9da6b)
The icon in the popup is indeed helpful, thanks for that idea!

Reply:
Hmm, should we do the same thing in the popup for modal filter icons then? Added as something to revisit in #30